diff --git a/src/cli/commands/monitor/index.ts b/src/cli/commands/monitor/index.ts
index 283fe871..0f7d025e 100644
--- a/src/cli/commands/monitor/index.ts
+++ b/src/cli/commands/monitor/index.ts
@@ -90,6 +90,10 @@ async function monitor(...args0: MethodArgs): Promise<any> {
     throw new Error('`--remote-repo-url` is not supported for container scans');
   }
 
+  if(options.longFormProjectName && !options.allProjects) {
+    throw new Error('`--longFormProjectName` is only compatible with `--all-projects`');
+  }
+
   apiTokenExists();
 
   let contributors: Contributors[] = [];
@@ -206,12 +210,18 @@ async function monitor(...args0: MethodArgs): Promise<any> {
 
           if (projectDeps.depGraph) {
             debug(`Processing ${projectDeps.depGraph.rootPkg.name}...`);
+            if(options.longFormProjectName){
+              options["project-name"] = projectDeps.depGraph.rootPkg.name + ':' +projectDeps.depGraph.rootPkg.version
+            }
             maybePrintDepGraph(options, projectDeps.depGraph);
             projectName = projectDeps.depGraph.rootPkg.name;
           }
 
           if (projectDeps.depTree) {
             debug(`Processing ${projectDeps.depTree.name}...`);
+            if(options.longFormProjectName){
+              options["project-name"] = projectDeps.depTree.name + ':' +projectDeps.depTree.version
+            }
             maybePrintDepTree(options, projectDeps.depTree);
             projectName = projectDeps.depTree.name;
           }
diff --git a/src/lib/types.ts b/src/lib/types.ts
index a5c773b4..97bf04d7 100644
--- a/src/lib/types.ts
+++ b/src/lib/types.ts
@@ -71,6 +71,7 @@ export interface Options {
 
 // TODO(kyegupov): catch accessing ['undefined-properties'] via noImplicitAny
 export interface MonitorOptions {
+  'longFormProjectName'?: boolean;
   id?: string;
   docker?: boolean;
   file?: string;
