diff --git a/src/cli/commands/monitor/index.ts b/src/cli/commands/monitor/index.ts
index 00e2bd29..ae4be89e 100644
--- a/src/cli/commands/monitor/index.ts
+++ b/src/cli/commands/monitor/index.ts
@@ -91,6 +91,10 @@ async function monitor(...args0: MethodArgs): Promise<any> {
     throw new Error('`--remote-repo-url` is not supported for container scans');
   }
 
+  if(options.longFormProjectName && !options.allProjects) {
+      throw new Error('`--longFormProjectName` is only compatible with `--all-projects`');
+  }
+
   apiTokenExists();
 
   let contributors: Contributors[] = [];
@@ -217,11 +221,17 @@ async function monitor(...args0: MethodArgs): Promise<any> {
           const projectName = getProjectName(projectDeps);
           if (projectDeps.depGraph) {
             debug(`Processing ${projectDeps.depGraph.rootPkg?.name}...`);
+            if(options.longFormProjectName){
+              options["project-name"] = projectDeps.depGraph.rootPkg.name + ':' +projectDeps.depGraph.rootPkg.version
+            }
             maybePrintDepGraph(options, projectDeps.depGraph);
           }
 
           if (projectDeps.depTree) {
             debug(`Processing ${projectDeps.depTree.name}...`);
+            if(options.longFormProjectName){
+              options["project-name"] = projectDeps.depTree.name + ':' +projectDeps.depTree.version
+            }
             maybePrintDepTree(options, projectDeps.depTree);
           }
 
diff --git a/src/lib/types.ts b/src/lib/types.ts
index 70147067..e2c42a24 100644
--- a/src/lib/types.ts
+++ b/src/lib/types.ts
@@ -77,6 +77,7 @@ export interface Options {
 
 // TODO(kyegupov): catch accessing ['undefined-properties'] via noImplicitAny
 export interface MonitorOptions {
+  'longFormProjectName'?: boolean;
   id?: string;
   docker?: boolean;
   file?: string;
